### PR TITLE
Remove some constructors

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: include/exiv2/metadatum.hpp [hunk 1]
```diff
@@ -264,6 +260,8 @@ class EXIV2API Metadatum {
     @brief Assignment operator. Protected so that it can only be used
            by subclasses but not directly.
    */
+  Metadatum() = default;
```
Comment (piponazo) [inline include/exiv2/metadatum.hpp:263]: Ey why did you move these constructors? Now the Doxygen groups does not make sense anymore. The creators are empty and the default constructor and the copy constructor are under the documentation for the assignment operator.

File: include/exiv2/metadatum.hpp [hunk 2]
```diff
@@ -80,6 +78,8 @@ class EXIV2API Key {
     @brief Assignment operator. Protected so that it can only be used
            by subclasses but not directly.
    */
+  Key() = default;
```
Comment (piponazo) [inline include/exiv2/metadatum.hpp:81]: Same comment here about moving the constructors and doxygen doc.

Comment (neheb) [reply include/exiv2/metadatum.hpp]: Fixed.

File: include/exiv2/metadatum.hpp (hunk 1 above)
Comment (neheb) [reply include/exiv2/metadatum.hpp]: Fixed.

**Comment by neheb** [comment]
out of curiosity, what happened to pvs studio and such?
